### PR TITLE
Fixing issue #153 (failing tests)

### PR DESCRIPTION
**Pull request by mpatsute**
cc @bajtos @raymondfeng @ritch @superkhau
Hey All, "npm test" was failing, I got it working after some fixes. Had to make a change in SwaggerRouter.ts and added some test related dependencies in package.json.
Please take a look, and merge, if everything looks ok to you.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,7 +18,12 @@
     "ts-node": "^2.0.0",
     "tsconfig-paths": "^2.1.0",
     "tslint": "^4.4.2",
-    "typescript": "^2.1.5"
+    "typescript": "^2.1.5",
+    "http-errors": "^1.6.1",
+    "path-to-regexp": "^1.7.0",
+    "should": "^11.2.1",
+    "sinon": "^2.1.0",
+    "supertest": "^3.0.0"
```
Comment (superkhau) [inline package.json]: this looks suspicious -- all the test deps should be inside packages/testlab

File: packages/loopback/lib/router/SwaggerRouter.ts [hunk 2]
```diff
@@ -197,8 +197,8 @@ class Endpoint {
 function buildOperationArguments(operationSpec: OperationObject, request: ParsedRequest,
   // tslint:disable-next-line:no-any
   pathParams: {[key: string]: string}): any[] {
-  const args = [];
-  
+  const args = new Array<string>();
```
Comment (superkhau) [inline packages/loopback/lib/router/SwaggerRouter.ts]: this looks suspicious too -- does `const args: Array<string> = []` work?

Comment (raymondfeng) [inline packages/loopback/lib/router/SwaggerRouter.ts]: You can use `const args: string[] = [];` or `const args = [] as string[]`;

Comment (superkhau) [inline packages/loopback/lib/router/SwaggerRouter.ts]: I good with both of your suggestions (or mine).

**Comment by mpatsute** [comment]
Fixed the PR based on review comments:
1. Removed newly added dependencies in the top level package.json
2. Fixed SwaggerRouter.ts, to declare type properly

**Comment by bajtos** [comment]
Landed, thank you @mpatsute for taking the initiative to fix this issue!

**Pull request merged**